### PR TITLE
Aberrant Organs - bio-engineer role, medbay map changes, surgery UI, RNG reduction

### PR DESCRIPTION
**Pull request by Pink-Chink**
### Moebius Bio-Engineer

Adds the Moebius Bio-Engineer, a new role for Moebius Medical for medical-adjacent research. The bio-engineer is given charge of genetics and visceral research (new name for organ laboratory). Starts with 35 BIO, 20 COG, 15 MEC. Starting gear includes an organ freezer instead of an advanced med kit. Science goggles are available to them from the loadout.

### Misc. Moebius Stuff

- Science goggles grant the ability to see organoid information.
- Genetics moved to Medical.
- Psychology moved to Science (there are plans for a rework).
- New disks for genetics and viscera machinery.
- Includes a partial re-map of medbay. Added disposal to operating theatre (not shown).

image
image
image
image
image

### Surgery UI

Adds modding buttons to surgery UI.

188351770-8d557b46-0dea-4b8b-a8df-f267684d5e57

### Misc. Changes

- Aberrant organ functions (organ tissues and organoid info) require 35 BIO and 10 COG to see on examine. Removal and attaching still require 40 BIO for 0% fail chance and 50 BIO for instant use. However, organoids won't be destroyed on failure.
- Organs don't rot in vending machines.
- Halved spawn rate of deep maint organ machines.
- Renamed regurgitator to disgorger. Genetics already used the name.
- Metabolic and endocrinal teratomas are separated by their input pools. No need to reroll to get the input you want.
- Dependent, enigmatic, and parasitic teratomas added to the pool of random drops. These are found in the medi-sarcophagus loot.
- Durable membrane size mod decreased from 0.5 to 0.25.
- Biostructure gel size mod decreased from 0.5 to 0.25.
- Hormone blood-clotting effect increased from 0.1 to 0.2.
- Adds parenchymal membrane, a mod that applies new organ efficiencies or increases existing ones. Found in the Bespoke Teratomas disk. Pygmy variant (lower efficiencies) found in the starting design of the organ fabricator.
- Reworks parasitic organoids. If you do not have melee armor on your hand when picking these up and your arm has free space, they will force their way into your body. Consider wearing gloves when working with unknown teratomas.
- Reduced BIO requirement for examining regular organs from 40 to 35 to make it more consistent with aberrant organs.

## Why It's Good For The Game

No more doctors taking up needed slots to go play with aberrant organs. Merging genetics with visceral research and placed under MoeMed creates a cleaner thematic separation between medical and science.

## Changelog
:cl:
add: Adds Moebius Bio-Engineer, a new MoeMed role for genetics + viscera
add: Added mod attach/remove buttons to organs in the surgery window
add: Added examine button to aberrant organs in the surgery window
add: Added dependent, enigmatic, and parasitic teratomas to the loot drop pool
add: Added Moebius genetics and viscera disks
add: Added parenchymal membrane
balance: Aberrant organ examine info requires 35 BIO and 10 COG
balance: Organoids don't have a chance to be destroyed on failed removal
balance: Durable membrane and biostructure gel size mod reduced to 0.25
balance: Hormone blood-clotting effect increased to 0.2
tweak: Reduced BIO req for regular organ info from 40 to 35
tweak: Genetics moved to medical, psych moved to science
tweak: Sci goggles allow user to see organoid information
tweak: Reagent-based teratomas are separated by their reagent pool
tweak: Parasitic organoids reworked
tweak: Renamed regurgitator to disgorger
tweak: Organs don't rot in vending machines
tweak: Halved spawn rate of deep maint organ machines
fix: Fixes disgorger substrate value going negative
fix: Adds stomach to the organ fab
/:cl:

### CODE REVIEW

**Comment by L74814** [comment]
yoo this is actual eris content.  I still don't understand what organs are so far but amazing. I hope you add the clearly canon new lore for the psychiatrist in the job description.

>less clueless eris player

**Comment by Pink-Chink** [comment]
nullAurelian has been working on a guide for the organs.

**Comment by Firefox13** [comment]
(image first, explanation for the image after it)
изображение
please change the tiles there so it looks a little more nicer
изображение
If you have shutters there might as well make windows reinforced
изображение
Might as well remove this appendix. Single-tile long stretches sometimes don't cut it

**Comment by Pink-Chink** [comment]
image
image
image

**Comment by maikilangiolo** [comment]
>starts with 40 BIO
Guh

**Comment by hex37** [comment]
Looking forward to pulling the fixes into sojourn

**Comment by Pink-Chink** [comment]
40 BIO because it’s weird to not have a way to view organoids unassisted at roundstart.

**Comment by Pink-Chink** [comment]
`Looking forward to pulling the fixes into sojourn `

I think the surgery UI and the reagent teratoma splitting are already merged.

**Comment by GrayRachnid** [comment]
Can you have someone make some changes to the base labcoat and doctor jumpsuit just to make the bioengineer look unique?

**Comment by GrayRachnid** [comment]
```
[17:57:26] Runtime in mob_helpers.dm,598: Cannot read /decl/move_intent/run (/decl/move_intent/run).flags
  proc name: skill to evade traps (/mob/proc/skill_to_evade_traps)
  usr: The mouse () (/mob/living/simple_animal/mouse)
  usr.loc: The underplating (80,122,4) (/turf/simulated/floor/plating/under)
  src: the mouse (/mob/living/simple_animal/mouse)
  src.loc: the underplating (80,122,4) (/turf/simulated/floor/plating/under)
  call stack:
  the mouse (/mob/living/simple_animal/mouse): skill to evade traps()
  the wire splicing (/obj/structure/wire_splicing): Crossed(the mouse (/mob/living/simple_animal/mouse))
  the mouse (/mob/living/simple_animal/mouse): forceMove(the underplating (80,122,4) (/turf/simulated/floor/plating/under), null, 0)
  the mouse (/mob/living/simple_animal/mouse): forceMove(the underplating (80,122,4) (/turf/simulated/floor/plating/under), null, 0)
  the mouse (/mob/living/simple_animal/mouse): forceMove(the underplating (80,122,4) (/turf/simulated/floor/plating/under), null, 0)
  the mouse (/mob/living/simple_animal/mouse): forceMove(the underplating (80,122,4) (/turf/simulated/floor/plating/under), null, 0)
  the open space (80,122,5) (/turf/simulated/open): fallThrough(the mouse (/mob/living/simple_animal/mouse))
  the mouse (/mob/living/simple_animal/mouse): init light()
  the mouse (/mob/living/simple_animal/mouse): New(the open space (80,122,5) (/turf/simulated/open))
  the mouse (/mob/living/simple_animal/mouse): New(the open space (80,122,5) (/turf/simulated/open))
  the mouse (/mob/living/simple_animal/mouse): New(the open space (80,122,5) (/turf/simulated/open))
  /datum/turf_initializer/mainte... (/datum/turf_initializer/maintenance): Initialize(the open space (80,122,5) (/turf/simulated/open))
  Second Section Deck 1 Port Mai... (/area/eris/maintenance/section2deck1port): Initialize(1)
  Atoms (/datum/controller/subsystem/atoms): InitAtom(Second Section Deck 1 Port Mai... (/area/eris/maintenance/section2deck1port), 0, /list (/list))
  Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null, null)
  Atoms (/datum/controller/subsystem/atoms): Initialize(646279)
  Master (/datum/controller/master): Initialize(10, 0, null)```

**Comment by garrulousTeamotei** [comment]
I'd drop the self medicated perk from the Bio-Engineer, and give them a new one that allows them to tell the function of any organoid regardless of bio level, give them the ability to successfully perform the laser scalpel check on organoids (but still keep the time to complete the check based of off bio). 
A bit too much overlap on the medical doctors skill points? Maybe make the time to complete the genetic mutation decryption based off of the Cog skill of the player, and computer components, like the ID hacking program from the Warez disk (easy to steal the code?). Make it punishing for low Cog skills, and move 10 of the bio-engineer's bio skill over to Cog. 
Have implanting the created organoids primarily apart of the medical doctors pervue?

**Comment by Pink-Chink** [comment]
>Can you have someone make some changes to the base labcoat and doctor jumpsuit just to make the bioengineer look unique?

I'm told Shaz is on it. As far as that runtime, it doesn't look like it's from anything I've touched, but we'll see next commit.

>I'd drop the self medicated perk from the Bio-Engineer, and give them a new one that allows them to tell the function of any organoid regardless of bio level, give them the ability to successfully perform the laser scalpel check on organoids (but still keep the time to complete the check based of off bio).

I'm not a fan of using perks to skip stat checks. I do like the idea of having a split stat requirement (BIO and COG) for organoids and making the Bio-Engi something like 30 BIO/20 COG or 35 BIO/15 COG. Puts him between the roboticist and doctor.

**Comment by Pink-Chink** [comment]
Flesh/peach uniform + sec HUD icon
image
image

**Comment by SirRichardFrancis** [comment]
> [17:57:26] Runtime in mob_helpers.dm,598: Cannot read /decl/move_intent/run (/decl/move_intent/run).flags


image

**Comment by Pink-Chink** [comment]
Tweaked the rooms a bit. See main post.

**Comment by garrulousTeamotei** [comment]
-The Bio-Engineer doesn't have access to the Moebius Doctor's Locker (Two are located in Visceral Research)
-Only Cognition is required to view the function of an organoid. Viewing the function of an implanted organoid requires the Cog+Bio combo. (Might just be moot with the needed 40+ Bio check to remove it in the first place)
-The Old Genetics lab still has Genetics lab access, not Psychiatrist's office. 
-Should a new Visceral access be made, giving that access to all Moebius Medical, and not Moebius Research? (like Chemistry)
-Depending on how much camera coverage of the new areas you want, the room with the organ fabricators only has the four eastern most tiles visible, and the locker portion of the break room is mostly outside camera range.
-The new Psychiatrist's office disposals ejects it's contents straight out into the adjacent primary hallway (Excellent feature for removing problem patients)
-Parasitic Organoids appear to be non-functional. Using the laser scalpel on it fails to remove the organoid from it's frame. holding the frame does not cause any organoid to be implanted into the holder, even when naked. This seems true of all parasitic organoids regardless of organ size. on separate notes, the base arm starts with .5 free space, which would make all with a larger size than that unable to be effective parasites. The parasitic organoid has a name and description of it being a parasite; maybe best to have it be seen and described as a teratoma (unknown), and leave the effects hidden in it's organoid descriptions for those with the needed Bio+Cog score, if the reason for adding the organoid is what I think it is, a trap.

**Comment by Pink-Chink** [comment]
Bugs should be fixed. Thanks for finding these.

Moved genetics access to med and made it the required access for Genetics + Visceral Research. 
The parasitic one will work as long as there is free space in the arm. Also, it only spawns from the medi-sarcophagus drops and those always appear as "teratoma (unknown)".

File: code/modules/surgery/mods.dm [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		if(limb.get_total_occupied_volume() + organ_size_delta > limb.max_volume)
+			to_chat(user, SPAN_WARNING("There isn't enough space in \the [limb] to apply \the [mod]."))
+			return FALSE
+
+	return organ.is_open()
+
+/datum/surgery_step/attach_mod/begin_step(mob/living/user, obj/item/organ/internal/organ, obj/item/mod)
+	var/obj/item/organ/external/limb = organ.get_limb()
+	if(limb)
+		organ.owner_custom_pain("Someone is digging into into your [limb.name]!", 1)
```
Comment (hyperioo) [inline code/modules/surgery/mods.dm:30]:
```suggestion
		organ.owner_custom_pain("Someone is digging into your [limb.name]!", 1)
```

File: code/modules/surgery/mods.dm [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	duration = 0
+	blood_level = 1
+
+/datum/surgery_step/remove_mod/can_use(mob/living/user, obj/item/organ/internal/organ, obj/item/tool)
+	return organ.is_open()
+
+/datum/surgery_step/remove_mod/begin_step(mob/living/user, obj/item/organ/internal/organ, obj/item/tool)
+	var/obj/item/organ/external/limb = organ.get_limb()
+	if(limb)
+		organ.owner_custom_pain("Someone is cutting into into your [limb.name]!", 1)
```
Comment (hyperioo) [inline code/modules/surgery/mods.dm:54]:
```suggestion
		organ.owner_custom_pain("Someone is cutting into your [limb.name]!", 1)
```

**Review by hyperioo — APPROVED** [review]
Map changes have several areas that I think would look better if they were merged, but nothing that cannot be fixed in another PR once we have a good look with ingame lightning.

**Pull request merged**